src/lib/utils/options/declaration.ts

```diff
@@ -238,6 +269,12 @@ export interface MapDeclarationOption<T> extends DeclarationOptionBase {
      * Optional override for the error reported when an invalid key is provided.
      */
     mapError?: string;
+
+    /**
+     * An optional validation function that validates a potential value of this option.
+     * The function must throw an Error if the validation fails and should do nothing otherwise.
+     */
+    validate?: (value: T) => void;
```

Comment:
I also don't think validate makes sense here... mapped option types are already constrained to a specific set of values.

Reply:
Same as for boolean option?